src/rules/unbound-method.ts

```diff
@@ -0,0 +1,111 @@
+import { TSESLint, TSESTree } from '@typescript-eslint/experimental-utils';
+import { createRule, isExpectCall, parseExpectCall } from './utils';
+
+const toThrowMatchers = [
+  'toThrow',
+  'toThrowError',
+  'toThrowErrorMatchingSnapshot',
+  'toThrowErrorMatchingInlineSnapshot',
+];
+
+const isJestExpectCall = (node: TSESTree.CallExpression) => {
+  if (!isExpectCall(node)) {
+    return false;
+  }
+
+  const { matcher } = parseExpectCall(node);
+
+  if (!matcher) {
+    return false;
+  }
+
+  return !toThrowMatchers.includes(matcher.name);
+};
+
+const baseRule = (() => {
```

Comment:
```suggestion
const baseRule: TSESLint.RuleModule<
      MessageIds,
      Options
    > | null = (() => {
```

rather than casting

Reply:
I'm happy to add the return type, but would prefer to keep the cast as it's narrowing an `any`, which otherwise is [an unsafe return](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unsafe-return.md)